### ISSUE TITLE
Allow adding extra information to `ipfs id` object

### ISSUE THREAD

**Issue by mitar** [ipfs/notes#227]
I have a web interface on my app on top of IPFS. I would like to add extra information to IPFS instances, so that I can use `ipfs id` to retrieve them. For example, a public URL for the web interface. So that other instances can direct users to peer's web interface.

**Comment by whyrusleeping** [comment]
@mitar so youre wanting something like `ipfs id QmsomePeer` and getting information that that peer has set on their ID 'extra info' field?

**Comment by mitar** [comment]
Exactly. So I would need two APIs, one to store this extra info. And one to then retrieve. For that I think existing `id` is the best, but it could also be something completely separate. I just see that currently `id` already returns various useful info.

**Comment by jbenet** [comment]
- yeah should be doable.
- and we should be able to mount protocols (register handlers) that show up there

**Comment by diasdavid** [comment]
What about moving this `id` command under `p2p` as it this info really comes from libp2p? Other commands are: dht, swarm, bootstrap and ping, essentially the network commands :) 

This creates a nice and obvious boundary between IPFS and libp2p features, making it more clear for users to report their issues, to find solutions, to know where issues might be coming from and so on.

**Comment by Kubuxu** [comment]
`ipfs p2p` was selected for stream mounting feature. It is implmented in go-ipfs:
```
USAGE
  ipfs p2p - Libp2p stream mounting.

  ipfs p2p

  Create and use tunnels to remote peers over libp2p
  
  Note: this command is experimental and subject to change as usecases and APIs are refined

SUBCOMMANDS
  ipfs p2p listener - P2P listener management.
  ipfs p2p stream   - P2P stream management.

Use 'ipfs p2p --help' for more information about this command.
```
but we can change it.

**Comment by lgierth** [comment]
> What about moving this id command under p2p as it this info really comes from libp2p? Other commands are: dht, swarm, bootstrap and ping, essentially the network commands :)

Seems unneccessary to me. Yes, arguably a bit tidier, but so many things that can go wrong with regard to backward-compat if we don't pay proper attention.
